1. Find all of the numbers from 1–1000 that are divisible by 8

In [1]:
li = [i for i in range(1,1001) if i%8 == 0]
li

[8,
 16,
 24,
 32,
 40,
 48,
 56,
 64,
 72,
 80,
 88,
 96,
 104,
 112,
 120,
 128,
 136,
 144,
 152,
 160,
 168,
 176,
 184,
 192,
 200,
 208,
 216,
 224,
 232,
 240,
 248,
 256,
 264,
 272,
 280,
 288,
 296,
 304,
 312,
 320,
 328,
 336,
 344,
 352,
 360,
 368,
 376,
 384,
 392,
 400,
 408,
 416,
 424,
 432,
 440,
 448,
 456,
 464,
 472,
 480,
 488,
 496,
 504,
 512,
 520,
 528,
 536,
 544,
 552,
 560,
 568,
 576,
 584,
 592,
 600,
 608,
 616,
 624,
 632,
 640,
 648,
 656,
 664,
 672,
 680,
 688,
 696,
 704,
 712,
 720,
 728,
 736,
 744,
 752,
 760,
 768,
 776,
 784,
 792,
 800,
 808,
 816,
 824,
 832,
 840,
 848,
 856,
 864,
 872,
 880,
 888,
 896,
 904,
 912,
 920,
 928,
 936,
 944,
 952,
 960,
 968,
 976,
 984,
 992,
 1000]

2. Find all of the numbers from 1–1000 that have a 6 in them

In [1]:
var1 = [i for i in range(1,1001) if '6' in str(i)]
var1


[6,
 16,
 26,
 36,
 46,
 56,
 60,
 61,
 62,
 63,
 64,
 65,
 66,
 67,
 68,
 69,
 76,
 86,
 96,
 106,
 116,
 126,
 136,
 146,
 156,
 160,
 161,
 162,
 163,
 164,
 165,
 166,
 167,
 168,
 169,
 176,
 186,
 196,
 206,
 216,
 226,
 236,
 246,
 256,
 260,
 261,
 262,
 263,
 264,
 265,
 266,
 267,
 268,
 269,
 276,
 286,
 296,
 306,
 316,
 326,
 336,
 346,
 356,
 360,
 361,
 362,
 363,
 364,
 365,
 366,
 367,
 368,
 369,
 376,
 386,
 396,
 406,
 416,
 426,
 436,
 446,
 456,
 460,
 461,
 462,
 463,
 464,
 465,
 466,
 467,
 468,
 469,
 476,
 486,
 496,
 506,
 516,
 526,
 536,
 546,
 556,
 560,
 561,
 562,
 563,
 564,
 565,
 566,
 567,
 568,
 569,
 576,
 586,
 596,
 600,
 601,
 602,
 603,
 604,
 605,
 606,
 607,
 608,
 609,
 610,
 611,
 612,
 613,
 614,
 615,
 616,
 617,
 618,
 619,
 620,
 621,
 622,
 623,
 624,
 625,
 626,
 627,
 628,
 629,
 630,
 631,
 632,
 633,
 634,
 635,
 636,
 637,
 638,
 639,
 640,
 641,
 642,
 643,
 644,
 645,
 646,
 647,
 648,
 649,
 650,
 651,
 652,
 653,
 654,
 655,


3. Count the number of spaces in a string

In [7]:
str1 = str1 = "Sandeep sir's teaching is very understandable"
x = len([i for i in str1 if i == ' '])
x

3

4. Remove all of the vowels in a string

In [4]:
str1 = "Sandeep sir's teaching is very understandable"
var2 = [' '.join([i for i in str1 if i not in ['a','e','i','o','u']])]
var2

["S n d p   s r ' s   t c h n g   s   v r y   n d r s t n d b l"]

5. Find all of the words in a string that are less than 5 letters

In [7]:
str1 = "Sandeep sir's teaching is very understandable"
var3 = [i for i in str1.split() if len(i)<5]
var3


['is', 'very']

6. Use a dictionary comprehension to count the length of each word in a sentence

In [9]:
str1 = "Sandeep sir's teaching is very understandable"
dic1 = {i:len(i) for i in str1.split()}
dic1

{'Sandeep': 7,
 "sir's": 5,
 'teaching': 8,
 'is': 2,
 'very': 4,
 'understandable': 14}

7. Use a nested list comprehension to find all of the numbers from 1–1000 that are divisible by single digit besides 1 (2–5)

In [131]:
var3 = [[j for j in range(2,6) if i % j == 0] for i in range(1,1001)]
var3

# PENDING !!!

[[],
 [2],
 [3],
 [2, 4],
 [5],
 [2, 3],
 [],
 [2, 4],
 [3],
 [2, 5],
 [],
 [2, 3, 4],
 [],
 [2],
 [3, 5],
 [2, 4],
 [],
 [2, 3],
 [],
 [2, 4, 5],
 [3],
 [2],
 [],
 [2, 3, 4],
 [5],
 [2],
 [3],
 [2, 4],
 [],
 [2, 3, 5],
 [],
 [2, 4],
 [3],
 [2],
 [5],
 [2, 3, 4],
 [],
 [2],
 [3],
 [2, 4, 5],
 [],
 [2, 3],
 [],
 [2, 4],
 [3, 5],
 [2],
 [],
 [2, 3, 4],
 [],
 [2, 5],
 [3],
 [2, 4],
 [],
 [2, 3],
 [5],
 [2, 4],
 [3],
 [2],
 [],
 [2, 3, 4, 5],
 [],
 [2],
 [3],
 [2, 4],
 [5],
 [2, 3],
 [],
 [2, 4],
 [3],
 [2, 5],
 [],
 [2, 3, 4],
 [],
 [2],
 [3, 5],
 [2, 4],
 [],
 [2, 3],
 [],
 [2, 4, 5],
 [3],
 [2],
 [],
 [2, 3, 4],
 [5],
 [2],
 [3],
 [2, 4],
 [],
 [2, 3, 5],
 [],
 [2, 4],
 [3],
 [2],
 [5],
 [2, 3, 4],
 [],
 [2],
 [3],
 [2, 4, 5],
 [],
 [2, 3],
 [],
 [2, 4],
 [3, 5],
 [2],
 [],
 [2, 3, 4],
 [],
 [2, 5],
 [3],
 [2, 4],
 [],
 [2, 3],
 [5],
 [2, 4],
 [3],
 [2],
 [],
 [2, 3, 4, 5],
 [],
 [2],
 [3],
 [2, 4],
 [5],
 [2, 3],
 [],
 [2, 4],
 [3],
 [2, 5],
 [],
 [2, 3, 4],
 [],
 [2],
 [3, 5],
 [2, 4]

8. Use list comprehension to create a list of cubes of the integers 1-10.

In [18]:
var4 = [i ** 3 for i in range(1,11)]
var4

[1, 8, 27, 64, 125, 216, 343, 512, 729, 1000]

9. Use list comprehension to make a list of the first letter of each word in the following list

In [50]:
wordList = ["this", "is", "an", "example"]

var5 = [ i[0] for i in wordList]
var5

['t', 'i', 'a', 'e']

10. Write a list comprehension that returns a list [0, 1, 4, 9, 16, ...., 625] starting from range(0, 26).

In [52]:
var6 = [i **2  for i in range(0,26)]
var6

[0,
 1,
 4,
 9,
 16,
 25,
 36,
 49,
 64,
 81,
 100,
 121,
 144,
 169,
 196,
 225,
 256,
 289,
 324,
 361,
 400,
 441,
 484,
 529,
 576,
 625]

11. A string is given:  msg = "happy birthday!"


Write a list comprehension that prints a list ['h', 'a', 'p', 'p', 'y', 'b', 'i', 'r', 't', 'h', 'd', 'a', 'y']

In [57]:
msg = "happy birthday!"

var7 = [i for i in msg if i.isalnum()]
var7

['h', 'a', 'p', 'p', 'y', 'b', 'i', 'r', 't', 'h', 'd', 'a', 'y']

12. Combining multiple lists into one:

list1 = [1, 2, 3]

list2 = [4, 5, 6]

list3 = [7, 8, 9]

In [59]:
list1 = [1, 2, 3]
list2 = [4, 5, 6]
list3 = [7, 8, 9]
var5 = [list1 + list2 + list3]
var5

[[1, 2, 3, 4, 5, 6, 7, 8, 9]]

13. Creating a list of tuples with pairs of elements from two lists:

list1 = [1, 2, 3]

list2 = ['a', 'b', 'c']

sample output :[(1, 'a'), (1, 'b'), (1, 'c'), (2, 'a'), (2, 'b'), (2, 'c'), (3, 'a'), (3, 'b'), (3, 'c')]

In [76]:
list1 = [1, 2, 3]
list2 = ['a', 'b', 'c']

var8 = [(i,j) for i in list1 for j in list2 ]
var8

# li = []
# for i in list1:
#     for j in list2:
#         li.append((i,j))
# print(li)
        

[(1, 'a'),
 (1, 'b'),
 (1, 'c'),
 (2, 'a'),
 (2, 'b'),
 (2, 'c'),
 (3, 'a'),
 (3, 'b'),
 (3, 'c')]

14. Creating a list of palindrome words:

words = ["level", "hello", "world", "racecar"]

In [84]:
words = ["level", "hello", "world", "racecar"]

var3 = [i for i in words if i == i[::-1]]
var3

# for i in words:
#     if i == i[::-1]:
#         print(i)

['level', 'racecar']

15. Combining elements of two lists into a single list:

list1 = [1, 2, 3]

list2 = ['a', 'b', 'c']

sample output : ['1a', '1b', '1c', '2a', '2b', '2c', '3a', '3b', '3c']

In [17]:
list1 = [1, 2, 3]
list2 = ['a', 'b', 'c']

var3 = [str(i)+j for i in list1 for j in list2]
var3       

['1a', '1b', '1c', '2a', '2b', '2c', '3a', '3b', '3c']

16. Filtering negative numbers and converting them to positive:

numbers = [1, -2, 3, -4, 5,4,5,6,3,6,-23,16,-14,31,25,13,-15]

In [19]:
numbers = [1, -2, 3, -4, 5,4,5,6,3,6,-23,16,-14,31,25,13,-15]

var9 = [abs(i) for i in numbers if i<0 ]
var9

[2, 4, 23, 14, 15]

17. Removing vowels from a list of words:

words = ["apple", "banana", "cherry"

In [84]:
words = ["apple", "banana", "cherry"]
# print(type(' '.join(words)))
# print(type(words))
# for i in str(words):
#     if i not in ['a','e','i','o','u']:
#         print(' '.join(i),end=' ')


var6 = [' '.join([i for i in str(words) if i not in ['a','e','i','o','u']])]
var6


["[ ' p p l ' ,   ' b n n ' ,   ' c h r r y ' ]"]

18. Creating a list of squares for even numbers and cubes for odd numbers:
    
numbers =[1, 2,3,4,5]

In [95]:
numbers =  [1,2,3,4,5]
var8 = [i**2 if i % 2 == 0 else i**3 for i in numbers]
var8

# for i in numbers:
#     if i % 2 == 0:
#         print(i**2)
#     else:
#         print(i**3)
   


[1, 4, 27, 16, 125]

19. Creating a list of unique elements with their counts:

numbers = [1, 2, 2, 3, 4, 4, 5]

In [111]:
numbers = [1, 2, 2, 3, 4, 4, 5]

var5 = {i: numbers.count(i) for i in set(numbers) if i in numbers}
var5

# for i in set(numbers):
#     if i in numbers:
#         print({i:numbers.count(i)})
        

{1: 1, 2: 2, 3: 1, 4: 2, 5: 1}

20. Finding the maximum length word in a list:

words = ["apple", "banana", "cherry", "date"]

In [128]:
words = ["apple", "banana", "cherry", "date"]

max_len= 0
for i in words:
    if len(i)>max_len:
        
        max_len+=len(i)
        print(i)

apple
banana


21. Reversing words in a list of sentences:

sentences = ["Hello world", "Python is great"]

In [130]:
sentences = ["Hello world", "Python is great"]

var6 = [i[::-1] for i in sentences]
var6

['dlrow olleH', 'taerg si nohtyP']